packages/workspace/src/workspace/path_index.ts

```diff
@@ -85,12 +85,28 @@ const getElementPathHints = (element: Element): Iterable<[string, Path[]]> => {
   return wu(_.entries(pathHints))
 }
 
+export const getElementsPathHints = (unmergedElements: Element[]): Iterable<[string, Path[]]> =>
+  wu(unmergedElements.map(getElementPathHints)).flatten(true)
+
+
 export const createPathIndex = (unmergedElements: Element[]): PathIndex => {
-  const pathHints = wu(unmergedElements.map(getElementPathHints)).flatten(true)
+  const pathHints = getElementsPathHints(unmergedElements)
   const pathIndex = new PathIndex(pathHints)
   return pathIndex
 }
 
+export const updatePathIndex = (current: PathIndex, unmergedElements: Element[],
+  servicesToMaintain: string[]): PathIndex => {
+  if (servicesToMaintain.length === 0) {
+    return createPathIndex(unmergedElements)
+  }
+  const oldPathHintsToMaintain = Array.from(current.entries()).filter(value =>
+    servicesToMaintain.includes(value[0].split(ElemID.NAMESPACE_SEPARATOR)[0]))
+  const pathIndex = new PathIndex(oldPathHintsToMaintain)
+  pathIndex.setAll(getElementsPathHints(unmergedElements))
+  return pathIndex
+}
```

Comment:
If we would pass the services to update this can be a bit simpler I think
```suggestion
export const updatePathIndex = (
  current: PathIndex,
  elements: Element[],
  servicesToUpdate: string[],
): PathIndex => {
  const pathHints = wu.chain(
    wu(current.entries())
      .filter(([id]) => !servicesToUpdate.includes(ElemID.fromFullName(id).adapter)),
    getElementsPathHints(unmergedElements),
  )
  return createPathIndex(pathHints)
}
```

Reply:
Well, there are some good ideas that I didn't know here, and I'll use them, but I don't see how reversing the services you get really helps you. All you really removed is the "if" at the beginning, but I want the "if", it's there for performance reason, so that on a full fetch we don't process the path index for no reason.